### PR TITLE
Apply formatter

### LINKED ISSUE #1751: Consistent indendation in Nemo
I think it is somewhat annoying that Nemo does not use a consistent indentation throughout the project. Could we agree on one specific indentation, and push that throughout the project?

### PR DESCRIPTION
**Pull request by joschmitt**
I ran the Julia formatter on everything.
We can of course merge other pull requests first since this one is easier to update.
We can also discuss/bikeshed whether people actually like the formatting defaults. I would also be happy with just consistent indentation everywhere.

Closes #1751.

### CODE REVIEW

File: src/HeckeMiscFiniteField.jl [hunk 1]
```diff
@@ -33,38 +33,49 @@ end
...
+  )
+  for j in 1:length(st)
+    ccall(
+      (:nmod_poly_set_coeff_ui, libflint),
+      Nothing,
+      (Ref{fqPolyRepFieldElem}, Int, UInt),
+      d,
+      j - 1,
+      st[j],
+    )
```
Comment (joschmitt) [inline src/HeckeMiscFiniteField.jl:75]: Can't say I'm a huge fan of these additional line breaks in the `ccall`s.

Comment (lgoettgens) [reply src/HeckeMiscFiniteField.jl:75]: I agree. Does it help to increase the allowed line length to something absurd like 200? I don't think there is any other option to keep these as one-liner, is there?

Comment (joschmitt) [reply src/HeckeMiscFiniteField.jl]: I adjusted the rules a bit. To me it looks better now and at least it is not creating ~30000 new lines anymore.

File: src/flint/fmpz.jl [hunk 2]
```diff
@@ -164,20 +163,20 @@ is_positive(n::ZZRingElem) = sign(Int, n) > 0
 Return `true` if $a$ fits into an `Int`, otherwise return `false`.
 """
 fits(::Type{Int}, a::ZZRingElem) = ccall((:fmpz_fits_si, libflint), Bool,
-                                   (Ref{ZZRingElem},), a)
+  (Ref{ZZRingElem},), a)
```
Comment (thofma) [inline src/flint/fmpz.jl:166]: This

File: src/flint/fmpz.jl [hunk 3]
```diff
@@ -164,20 +163,20 @@ is_positive(n::ZZRingElem) = sign(Int, n) > 0
...
 
 @doc raw"""
     fits(::Type{UInt}, a::ZZRingElem)
 
 Return `true` if $a$ fits into a `UInt`, otherwise return `false`.
 """
 fits(::Type{UInt}, a::ZZRingElem) = a < 0 ? false :
-              ccall((:fmpz_abs_fits_ui, libflint), Bool, (Ref{ZZRingElem},), a)
+                                    ccall((:fmpz_abs_fits_ui, libflint), Bool, (Ref{ZZRingElem},), a)
```
Comment (thofma) [inline src/flint/fmpz.jl:174]: And this, does not make any sense

File: src/flint/fmpz.jl [hunk 4]
```diff
@@ -227,32 +226,32 @@ iseven(a::ZZRingElem) = ccall((:fmpz_is_even, libflint), Cint, (Ref{ZZRingElem},
...
 
 AbstractAlgebra.is_syntactic_one(x::ZZRingElem) = isone(x)
 
 AbstractAlgebra.is_syntactic_zero(x::ZZRingElem) = iszero(x)
 
 function AbstractAlgebra.print_obj(S::AbstractAlgebra.printer, mi::MIME,
-                                               obj::ZZRingElem, left::Int, right::Int)
-   AbstractAlgebra.print_integer_string(S, mi, string(obj), left, right)
+  obj::ZZRingElem, left::Int, right::Int)
+  AbstractAlgebra.print_integer_string(S, mi, string(obj), left, right)
```
Comment (thofma) [inline src/flint/fmpz.jl:241]: This looks bad

**Comment by thofma** [comment]
I welcome the change to use two spaces, but the automatic things are suboptimal, see my comments.

File: .JuliaFormatter.toml [hunk 5]
```diff
@@ -6,3 +6,5 @@ import_to_using = false
 indent = 2
 short_to_long_function_def = false
 always_for_in = true
+join_lines_based_on_source = true
```
Comment (lgoettgens) [inline .JuliaFormatter.toml:9]:
```suggestion
```
I think just having the overlong margin should be enough to fix the call issues. Having this one here is the root of some of the problems @thofma pointed out.
@joschmitt If you apply this change, please start again with the unformatted version. Repeated formatting with different configs may change the result over time.

File: src/flint/fmpz.jl [hunk 6]
```diff
@@ -306,64 +302,56 @@ round(::Type{ZZRingElem}, x::ZZRingElem) = x
...
-    else
-       ccall((:fmpz_divexact, libflint), Nothing,
-                           (Ref{ZZRingElem}, Ref{ZZRingElem}, Ref{ZZRingElem}), z, x, y)
-    end
-    return z
+  iszero(y) && throw(DivideError())
+  z = ZZRingElem()
+  if check
+    r = ZZRingElem()
+    ccall((:fmpz_tdiv_qr, libflint), Nothing, (Ref{ZZRingElem}, Ref{ZZRingElem}, Ref{ZZRingElem}, Ref{ZZRingElem}), z, r, x, y)
```
Comment (joschmitt) [inline src/flint/fmpz.jl:343]: Without the `join_lines_based_on_source` it removes newlines where the hand-formatted code deliberately had them. I don't like it.

**Comment by thofma** [comment]
But now many lines are longer than 80 characters.

**Comment by joschmitt** [comment]
I have the feeling that this automatic formatting is not converging... is anybody using this functionality in real life?

**Pull request closed without merging**